cSploit/src/org/csploit/android/core/System.java

```diff
@@ -319,16 +312,20 @@ public static void initCore() throws DaemonException, SuException {
     if(!Client.isConnected()) {
       if(!Client.Connect(socket_path)) {
         startCoreDaemon();
-        if (!Client.Connect(socket_path))
+        if (!Client.Connect(socket_path)) {
+          mCoreInitialized = false;
```

Comment:
`mCoreInitialized` is already false, give a look at the first statement of this function :wink: 
or maybe you found some strange race condition ?